### PR TITLE
Improve signup flow for early access to Khoj cloud

### PR DESCRIPTION
**Pull request by debanjum**
- Move early access signup to top of landing page. Earlier it was under /waitlist route only
- Move waitlist into component instead of standalone page. 
   Still keep /waitlist route available for direct sharing but not exposed via navigation bar
- Request only email address to signup, no need to fill users industry to signup
- Share link Google survey after early access signup complete
   The short survey currenty contains 3 questions including users industry

### CODE REVIEW

File: src/components/Waitlist.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	const SignupSchema = Yup.object().shape({
+		email: Yup.string().email('Invalid email').required('Required'),
+	});
+
+	const APIURL = process.env.NODE_ENV == 'production' ? "https://lantern.khoj.dev" : 'http://localhost:5000';
+
+	return (
+			<div className='signup-bar'>
+				<Formik
+					initialValues={{ email: '', interest: '' }}
```
Comment (sabaimran) [inline src/components/Waitlist.tsx:21]: We'll need to migrate the back-end schema before merging this.

File: src/styles/Waitlist.css [hunk 2]
```diff
@@ -1,23 +1,67 @@
...
+.signup-bar {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100%;
+    background-color: rgba(205, 205, 205, 0.7);
+    padding: 0px;
+    margin: 0px;
+    box-sizing: border-box;
+    z-index: 9999;
```
Comment (sabaimran) [inline src/styles/Waitlist.css:12]: What's the benefit to having the z-index placed in this way? This always ensures that it comes to the front of the page?

File: src/styles/Waitlist.css [hunk 3]
```diff
@@ -1,23 +1,67 @@
 @import url(colors.css);
 
-div.waitlist-form {
-    margin-left: 40%;
-    margin-right: 40%;
+.signup-bar {
```
Comment (sabaimran) [inline src/styles/Waitlist.css:3]: For consistency, I think it's better to maintain the `waitlist` nomenclature, as we'll predictably have a separate sign-up flow later.

File: src/components/Waitlist.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+					validationSchema={SignupSchema}
+					onSubmit={(values, { setSubmitting }) => {
+						setTimeout(() => {
```
Comment (sabaimran) [inline src/components/Waitlist.tsx:24]: As part of this iteration, let's also add some text about what we're making/why users should sign up, and (potentially) make a shareable link that goes straight to a sign-up page.

File: src/styles/Waitlist.css (hunk 2 above)
Comment (debanjum) [reply src/styles/Waitlist.css:12]: Yes, exactly. For the notice to be on top of all other elements on the page

File: src/components/Waitlist.tsx (hunk 1 above)
Comment (debanjum) [reply src/components/Waitlist.tsx:21]: This should be done now?

**Pull request merged**